src/main/java/picard/arrays/illumina/CompareGtcFiles.java

```diff
@@ -0,0 +1,236 @@
+
+/**
+ * A simple tool to compare two Illumina GTC files.
+ */
+
+@CommandLineProgramProperties(
+        summary = CompareGtcFiles.USAGE_DETAILS,
+        oneLineSummary = "Compares two GTC files.",
+        programGroup = picard.cmdline.programgroups.GenotypingArraysProgramGroup.class
+)
+public class CompareGtcFiles extends CommandLineProgram {
+
+    static final String USAGE_DETAILS =
+            "CompareGtcFiles takes two Illumina GTC file and compares their contents to ensure that fields expected to be the same " +
+                    "are in fact the same.  This will exclude any variable field, such as a date. " +
+                    "The GTC files must be generated on the same chip type. " +
+                    "<h4>Usage example:</h4>" +
+                    "<pre>" +
+                    "java -jar picard.jar CompareGtcFiles \\<br />" +
+                    "      INPUT=input1.gtc \\<br />" +
+                    "      INPUT=input2.gtc \\<br />" +
+                    "      ILLUMINA_NORMALIZATION_MANIFEST=chip_name.bpm.csv \\<br />" +
+                    "</pre>";
+
+    private static Log log = Log.getInstance(CompareGtcFiles.class);
+
+    @Argument(shortName = StandardOptionDefinitions.INPUT_SHORT_NAME,
+            doc = "GTC input files to compare.",
+            minElements = 2,
+            maxElements = 2)
+    public List<File> INPUT;
+
+    @Argument(shortName = "NORM_MANIFEST", doc = "An Illumina bead pool manifest (a manifest containing the Illumina normalization ids) (bpm.csv)")
+    public File ILLUMINA_NORMALIZATION_MANIFEST;
+
+    private List<String> errors = new ArrayList<>();
+
+    //ignored methods
+    private static final List<String> IGNORED_METHODS = new ArrayList<>();
+
+    static {
+        IGNORED_METHODS.add("getClass");
+        IGNORED_METHODS.add("getAutoCallDate");
+        IGNORED_METHODS.add("getImagingDate");
+        //This is the number of TOC entries. It will be different with different versions.
+        IGNORED_METHODS.add("getNumberOfEntries");
+        //We don't inject these in our gtcs so they will always be blank and so we don't bother comparing.
+        IGNORED_METHODS.add("getSampleName");
+        IGNORED_METHODS.add("getSamplePlate");
+        IGNORED_METHODS.add("getSampleWell");
+    }
+
+    @Override
+    protected int doWork() {
+        IOUtil.assertFilesAreReadable(INPUT);
+
+        try {
+            InfiniumNormalizationManifest infiniumNormalizationManifest
+                    = new InfiniumNormalizationManifest(ILLUMINA_NORMALIZATION_MANIFEST);
+            InfiniumGTCFile gtcFileOne = new InfiniumGTCFile(new DataInputStream(new FileInputStream(INPUT.get(0))), infiniumNormalizationManifest);
+            InfiniumGTCFile gtcFileTwo = new InfiniumGTCFile(new DataInputStream(new FileInputStream(INPUT.get(1))), infiniumNormalizationManifest);
+            compareGTCFiles(gtcFileOne, gtcFileTwo);
+
+            //report errors and exit 1 if any are detected.
```

Comment:
The java convention is to put a space after `//`, it would be nice to update the comments in this file to follow that standard. Also in cases like this, where the comment is a punctuated sentence, it should start with a capital letter.
```suggestion
            // Report errors and exit 1 if any are detected.
```

Reply:
yup.  I blame Jay.